src/Compilers/Core/Portable/CodeGen/ILBuilderEmit.cs

```diff
@@ -43,9 +43,14 @@ internal void EmitToken(string value)
             this.GetCurrentWriter().WriteUInt32(token);
         }
 
-        internal void EmitToken(Microsoft.Cci.IReference value, SyntaxNode syntaxNode, DiagnosticBag diagnostics)
+        internal void EmitToken(Microsoft.Cci.IReference value, SyntaxNode syntaxNode, DiagnosticBag diagnostics, bool encodeAsRawToken = false)
         {
             uint token = module?.GetFakeSymbolTokenForIL(value, syntaxNode, diagnostics) ?? 0xFFFF;
+            // Setting the high bit indicates that the token value is to be interpreted literally rather than as a handle.
+            if (encodeAsRawToken)
+            {
+                token |= 0x80000000;
```

Comment:
I'm not sure what a meaningful assertion would be here. The line above, which apparently uses 0xFFFF as an invalid token value, troubles me because 65535 seems like too low a limit on the number of valid tokens. Why isn't the invalid value 0xFFFFFF or 0xFFFFFFFF? @tmat ? @VSadov ?